indico/htdocs/js/indico/jquery/rulelistwidget.js

```diff
@@ -0,0 +1,213 @@
+                    self.element.trigger('change');
+                });
+
+            if (condition.required) {
+                $select.prepend($('<option disabled selected>', {
+                    value: ''
+                }).text($T.gettext('Select an option')));
+            }
+
+            var $html = $('<div>', {
+                class: 'rule-condition'
+            }).attr('data-condition', condition.value);
+
+            if (condition.labelText) {
+                $html.append($('<label>').text(condition.labelText));
+            }
+            return $html.append($select);
+        },
+
+        _optionSelected: function(ruleContext, condition, newOption) {
+            ruleContext.rule[condition.value] = _listValue(newOption);
+            this._checkConditionIncompatibility(ruleContext);
+            this.element.trigger('change');
+        },
+
+        _onChange: function() {
+            // Simplify rule data by remove "field: any"
+
+            this.ruleData.forEach(function(rule) {
+                _.keys(rule).forEach(function(k) {
+                    if (rule[k] === '*') {
+                        delete rule[k];
+                    }
+                });
+            });
+
+            this.element.val(JSON.stringify(this.ruleData));
+        },
+
+        _checkConditionIncompatibility: function(ruleContext) {
+            ruleContext.$element.find('.rule-condition').show();
+
+            this.conditionSpec.forEach(function(condition) {
+                var $condition = ruleContext.$element.find('.rule-condition[data-condition=' + condition.value + ']');
+                var newOption = $condition.find('select').val();
+
+                if (condition.compatibleWith) {
+                    var compatibleConditions = condition.compatibleWith[newOption];
+                    ruleContext.$element.find('.rule-condition').each(function() {
+                        var $this = $(this);
+                        var conditionName = $this.data('condition');
+                        if (this === $condition.get(0)) {
+                            return;
+                        }
+                        if (_.contains(compatibleConditions, conditionName)) {
+                            $this.show();
+                        } else {
+                            $this.hide();
+                            delete ruleContext.rule[conditionName];
+                        }
+                    });
+                }
+            });
+        },
+
+        _setConditionOption: function(ruleContext, conditionName, value) {
+            var $condition = ruleContext.$element.find('.rule-condition[data-condition=' + conditionName + ']');
+            var $select = $condition.find('select');
+
+            if (value === undefined) {
+                value = this.options.conditionChoices[conditionName].options[0][0];
+            }
+
+            // will implicitly call _optionSelected(...)
```

Comment:
I believe that's a leftover of a previous incarnation. I'll remove that
comment.

On Wed, 7 Sep 2016, 16:51 Adrian, <notifications@github.com> wrote:

> In indico/htdocs/js/indico/jquery/rulelistwidget.js
> <https://github.com/indico/indico/pull/2595#discussion_r77837767>:
>
> > +                            delete ruleContext.rule[conditionName];
> > +                        }
> > +                    });
> > +                }
> > +            });
> > +        },
> > +
> > +        _setConditionOption: function(ruleContext, conditionName, value) {
> > +            var $condition = ruleContext.$element.find('.rule-condition[data-condition=
>
> ' + conditionName + ']');
> > +            var $select = $condition.find('select');
> > +
> > +            if (value === undefined) {
> > +                value = this.options.conditionChoices[conditionName].options[0][0];
> > +            }
> > +
> > +            // will implicitly call _optionSelected(...)
>
> how? AFAIK .val() doesn't trigger any events
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/indico/indico/pull/2595/files/a503f55dc6f738b72d5a14a8bee21c03f762ba26#r77837767>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAKiyTBHehLu0gONnsPsMa2B7QdhEdLks5qns9XgaJpZM4J26K0>
> .
>